src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -224,6 +221,14 @@ def validate(self, node, options=None):
                             "are permitted to be Ranges but found {0} at "
                             "index {1}.".format(type(index).__name__, count+2))
 
+        if len(vector.symbol.shape) > 1 and not vector.children:
+            # If the vector has no children then it is a reference. If
+            # it is a reference then the number of arguments must be
+            # 1.
+            raise TransformationError(
+                "Expected 2nd child of a MATMUL BinaryOperation to have 1 "
+                "dimension, but found '{0}'."
+                "".format(len(matrix.symbol.shape)))
```

Comment:
Should that be `vector.symbol.shape`?

Reply:
Yes, well spotted. I guess the test should have picked it up but it is simple to use the same field for both arguments and this means you end up with the same error if the first argument or second argument is used. I've updated the test so it can't happen again.